app/services/puppet_fact_parser.rb

```diff
@@ -40,14 +40,20 @@ def operatingsystem
     end
     if os.description.blank?
       if os_name == 'SLES'
-        os.description = os_name + ' ' + orel.gsub!('.', ' SP')
+        os.description = os_name + ' ' + orel.gsub('.', ' SP')
       elsif facts[:lsbdistdescription]
         family = os.deduce_family || 'Operatingsystem'
         os.description = family.constantize.shorten_description facts[:lsbdistdescription]
       end
     end
-    os.save!
-    os
+
+    if os.new_record?
+      os.save!
```

Comment:
I was trying to avoid doing the extra .find call if the OS already existed. If it's a new record then its class will be Operatingsystem with `family` set to say, `Debian` so has to be reloaded to be the right class.

If the OS already existed and we're just updating it, the family will already be known, it'll be a `Debian` class and so saving it is enough - it doesn't need to reload from the DB, saving a call.